lib/ddtrace/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,75 @@
+require 'faraday'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/distributed'
+
+module Datadog
+  module Contrib
+    module Faraday
+      # Middleware implements a faraday-middleware for ddtrace instrumentation
+      class Middleware < ::Faraday::Middleware
+        DEFAULT_ERROR_HANDLER = lambda do |env|
+          Ext::HTTP::ERROR_RANGE.cover?(env[:status])
+        end
+
+        DEFAULT_OPTIONS = {
```

Comment:
👍  we can use this approach in general, but at some point we may design a common way (part of the `Pin` or `Monkey` or whatever) to update configurations for each integration. In that way we can standardize the way to change the integration behavior. For now, all good here.